packages/protocol-kit/src/utils/eip-712/encode.ts

```diff
@@ -0,0 +1,211 @@
+import { EIP712TypedData, TypedDataTypes, TypedMessageTypes } from 'packages/safe-core-sdk-types'
+import {
+  keccak256,
+  concat,
+  AbiParameter,
+  encodeAbiParameters,
+  getTypesForEIP712Domain,
+  validateTypedData,
+  hashDomain,
+  toHex,
+  Hex,
+  HashTypedDataParameters
+} from 'viem'
+import { asHex } from '../types'
+
+// This whole file was copied (and softly adapted) from viem in order to expose the function that provides just the encoding.
```

Comment:
I saw this was done to avoid to break the types in the current iteration. This was not explicitly explained here, and that this piece of code should be removed as soon as possible. Please add some comment signaling this.